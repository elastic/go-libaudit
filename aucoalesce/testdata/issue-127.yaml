---

tests:
  multiple_avc_messages: |
    type=AVC msg=audit(1668179838.476:649407): avc:  denied  { search } for  pid=4059486 comm="cephcsi" name="crypto" dev="proc" ino=475090959 scontext=system_u:system_r:svirt_lxc_net_t:s0:c222,c955 tcontext=system_u:object_r:sysctl_crypto_t:s0 tclass=dir permissive=1
    type=AVC msg=audit(1668179838.476:649407): avc:  denied  { read } for  pid=4059486 comm="cephcsi" name="fips_enabled" dev="proc" ino=475090960 scontext=system_u:system_r:svirt_lxc_net_t:s0:c222,c955 tcontext=system_u:object_r:sysctl_crypto_t:s0 tclass=file permissive=1
    type=AVC msg=audit(1668179838.476:649407): avc:  denied  { open } for  pid=4059486 comm="cephcsi" path="/proc/sys/crypto/fips_enabled" dev="proc" ino=475090960 scontext=system_u:system_r:svirt_lxc_net_t:s0:c222,c955 tcontext=system_u:object_r:sysctl_crypto_t:s0 tclass=file permissive=1
