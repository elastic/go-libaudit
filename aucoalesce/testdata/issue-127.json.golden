[
  {
    "test_name": "multiple_avc_messages",
    "event": {
      "@timestamp": "2022-11-11T15:17:18.476Z",
      "sequence": 649407,
      "category": "mac-decision",
      "record_type": "avc",
      "result": "unknown",
      "session": "",
      "summary": {
        "actor": {
          "primary": "system_u:system_r:svirt_lxc_net_t:s0:c222,c955"
        },
        "action": "violated-selinux-policy",
        "object": {
          "primary": "system_u:object_r:sysctl_crypto_t:s0"
        },
        "how": "cephcsi"
      },
      "user": {},
      "process": {
        "pid": "4059486",
        "name": "cephcsi"
      },
      "data": {
        "dev": "proc",
        "path": "/proc/sys/crypto/fips_enabled",
        "permissive": "1",
        "scontext": "system_u:system_r:svirt_lxc_net_t:s0:c222,c955",
        "seresult": "denied",
        "tcontext": "system_u:object_r:sysctl_crypto_t:s0"
      },
      "avc_data": [
        {
          "comm": "cephcsi",
          "dev": "proc",
          "ino": "475090959",
          "name": "crypto",
          "permissive": "1",
          "pid": "4059486",
          "scontext": "system_u:system_r:svirt_lxc_net_t:s0:c222,c955",
          "seperms": "search",
          "seresult": "denied",
          "tclass": "dir",
          "tcontext": "system_u:object_r:sysctl_crypto_t:s0"
        },
        {
          "comm": "cephcsi",
          "dev": "proc",
          "ino": "475090960",
          "name": "fips_enabled",
          "permissive": "1",
          "pid": "4059486",
          "scontext": "system_u:system_r:svirt_lxc_net_t:s0:c222,c955",
          "seperms": "read",
          "seresult": "denied",
          "tclass": "file",
          "tcontext": "system_u:object_r:sysctl_crypto_t:s0"
        },
        {
          "comm": "cephcsi",
          "dev": "proc",
          "ino": "475090960",
          "path": "/proc/sys/crypto/fips_enabled",
          "permissive": "1",
          "pid": "4059486",
          "scontext": "system_u:system_r:svirt_lxc_net_t:s0:c222,c955",
          "seperms": "open",
          "seresult": "denied",
          "tclass": "file",
          "tcontext": "system_u:object_r:sysctl_crypto_t:s0"
        }
      ],
      "ecs": {
        "event": {},
        "user": {
          "effective": {},
          "target": {},
          "changes": {}
        },
        "group": {}
      }
    },
    "warnings": [
      "duplicate key (ino) from AVC message",
      "duplicate key (ino) from AVC message",
      "duplicate key (name) from AVC message",
      "duplicate key (seperms) from AVC message",
      "duplicate key (seperms) from AVC message",
      "duplicate key (tclass) from AVC message",
      "duplicate key (tclass) from AVC message",
      "failed to set object secondary using keys=[tclass] because they were not found"
    ]
  }
]
